### PR TITLE
Remove definition of NULL, rely on stddef.h.

### PR DESCRIPTION
**Pull request by Axel-Naumann**

### CODE REVIEW

**Comment by vgvassilev** [comment]
Could we get rid of the other `git grep 'define NULL` in the codebase, too?

File: core/base/inc/RtypesCore.h [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 #include "RConfig.h"
 
-#include <stddef.h>
+#include <stddef.h> // size_t, NULL
```
Comment (amadio) [inline core/base/inc/RtypesCore.h]: Why not `#include <cstddef>`?

**Comment by Axel-Naumann** [comment]
a) because that's out of scope for this PR (I did not add the `#include`)
b) because we need `size_t`, not `std::size_t`.

**Comment by Axel-Naumann** [comment]
@vgvassilev none of those are visible to users. But be my guest...

**Comment by amadio** [comment]
> a) because that's out of scope for this PR (I did not add the #include)

Since `cstddef` usually includes `stddef.h` and adds a bit of C++ extras, there's no harm in changing it to the C++ header.

> b) because we need size_t, not std::size_t.

Even though the standard says that only `std::size_t` needs to be defined, `size_t` is defined anyway, because `cstddef` usually includes `stddef.h` before adding the C++ stuff on top.

This is what I see in GCC
```cpp
#ifndef _GLIBCXX_CSTDDEF
#define _GLIBCXX_CSTDDEF 1

#pragma GCC system_header

#include <bits/c++config.h>
#include <stddef.h>

#if __cplusplus >= 201103L
namespace std
{
  // We handle size_t, ptrdiff_t, and nullptr_t in c++config.h.
  using ::max_align_t;
}
#endif

#endif // _GLIBCXX_CSTDDEF
```
and Boost
```cpp
#ifndef __CSTDDEF_HEADER
#define __CSTDDEF_HEADER

#include <stddef.h>

namespace std {
  using ::ptrdiff_t;
  using ::size_t;
}

#endif // CSTDDEF_HEADER
```

**Comment by Axel-Naumann** [comment]
Changing code based on heuristics and guesses ("usually", "This is what I see in GCC") is not a stable approach to coding :-)

We need to write code that works on all compilers now and in 10 years. cstddef is meant to provide `std::size_t`, and we had cases in the past where `size_t` was indeed not available after including cstddef. So no, I'm not in favor of the change you propose, it's very different in personality from what I am changing here, and so *at least* requires motivation and a different PR.

**Pull request merged**